pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -31,21 +31,21 @@
 from pennylane.typing import ResultBatch
 
 
-def _create_transform_primitive(name):
+@functools.lru_cache
+def _create_transform_primitive():
     try:
         # pylint: disable=import-outside-toplevel
         from pennylane.capture.custom_primitives import QmlPrimitive
     except ImportError:
         return None
 
-    transform_prim = QmlPrimitive(name + "_transform")
+    transform_prim = QmlPrimitive("transform")
     transform_prim.multiple_results = True
     transform_prim.prim_type = "transform"
 
+    # pylint: disable=too-many-arguments, disable=unused-argument
     @transform_prim.def_impl
-    def _impl(
-        *all_args, inner_jaxpr, args_slice, consts_slice, targs_slice, tkwargs
-    ):  # pylint: disable=unused-argument
+    def _impl(*all_args, inner_jaxpr, args_slice, consts_slice, targs_slice, tkwargs, transform):
```

Comment:
For the sake of readability, is it worth having `name` as a kwarg to the primitive?